Index: src/ddscf/movecs_pr_anal.F
  ===================================================================
--- src/ddscf/movecs_pr_anal.F        (revision 24242)
+++ src/ddscf/movecs_pr_anal.F        (working copy)
@@ -94,14 +94,23 @@ c if there is no title sent, don't print
          write(LuOut,*)
 c
  110     continue
+
+c the next two lines are added to have full output (Hubert)
-          do i = ilo, ihi
+          do i = 1, nmo
+ c  changed to print all orbitals (Hubert)
 c
             call ga_get(g_vecs, 1, nbf, i, i, dbl_mb(k_vecs), 1)
 c
 c     Identify significant coefficients and sort by size
 c
+
+c  changed to print all orbitals (Hubert)
+c            thresh=0.0d0     
+
             n = 0
-            cur_thresh = thresh
+c            cur_thresh = thresh    chhanged (Hubert)
+            cur_thresh = 0.0d0
  111        do j = 0, nbf-1
                if (abs(dbl_mb(k_vecs+j)).ge.cur_thresh) then
                   int_mb(k_list + n) = j
@@ -196,14 +205,15 @@ c
  22         format(1x,2('  Bfn.  Coefficient  Atom+Function  ',5x))
             write(LuOut,23)
  23         format(1x,2(' ----- ------------  ---------------',5x))
-            do klo = 0, min(n-1,9), 2
+            do klo = 0, min(n-1,30009), 2
                khi = min(klo+1,n-1)
                write(LuOut,2) (
      $              int_mb(k_list+k)+1, 
      $              dbl_mb(k_vecs+int_mb(k_list+k)),
      $              (byte_mb(k_tags+int_mb(k_list+k)*16+m),m=0,15), 
      $              k = klo,khi)
- 2             format(1x,2(i5,2x,f12.6,2x,16a1,4x))
+c 2             format(1x,2(i5,2x,f12.1,2x,16a1,4x))  changed (Hubert) 
+ 2             format(1x,2(i5,2x,g20.12,2x,16a1,4x))
             enddo
             write(LuOut,*)
          enddo
